Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/preprocessor/transform/DifferenceTransformer.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.parkservices.preprocessor.transform;
+
+import java.util.Arrays;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import com.amazon.randomcutforest.parkservices.statistics.Deviation;
+import com.amazon.randomcutforest.returntypes.RangeVector;
+
+@Getter
+@Setter
+public class DifferenceTransformer extends WeightedTransformer {
+
+    public DifferenceTransformer(double[] weights, Deviation[] deviation) {
+        super(weights, deviation);
+    }
+
+    /**
+     * note that weight == 0, would produce 0 values in the inversion
+     *
+     * @param values        what the RCF would like to observe
+     * @param previousInput what was the real (or previously imputed) observation
+     * @return the observations that would (approximately) transform to values[]
+     */
+    public double[] invert(double[] values, double[] previousInput) {
```

Comment:
minor: is this an inherited method? can we put the `@Override` annotation on it?

Reply:
Done. Also fixed for the other transforms.